streampark-common/src/main/scala/org/apache/streampark/common/util/CompletableFutureUtils.scala

```diff
@@ -56,7 +56,12 @@ object CompletableFutureUtils {
       unit: TimeUnit,
       handle: JavaFunc[T, T],
       exceptionally: JavaFunc[Throwable, T]): CompletableFuture[T] = {
-    future.applyToEither(setTimeout(timeout, unit), handle).exceptionally(exceptionally)
+    future
+      .applyToEither(setTimeout(timeout, unit), handle)
+      .exceptionally(exceptionally)
+      .whenComplete(new BiConsumer[T, Throwable]() {
+        override def accept(t: T, u: Throwable): Unit = future.cancel(true)
```

Comment:
Could you please add some unit tests to check the future is canceled after timeout?

Reply:
> Could you please add some unit tests to check the future is canceled after timeout?

hi @1996fanrui :

whenComplete is the completion of the call, run after future successful and exceptionally(Include timeout)

Reply:
I understand the code, I just mean we should add a unit test.

Reply:
> I understand the code, I just mean we should add a unit test.

okay. I provide this testcase